src/adaptations/device-layer/EventLogging.cpp

```diff
@@ -35,21 +35,22 @@ namespace Weave {
 namespace DeviceLayer {
 namespace Internal {
 
+/* BSS segment is limited on some device, provides an option to allocate large memory buffer on heap */
+#ifdef WEAVE_DEVICE_CONFIG_EVENT_LOGGING_BUFFER_STATIC
 uint64_t gCritEventBuffer[RoundUp(WEAVE_DEVICE_CONFIG_EVENT_LOGGING_CRIT_BUFFER_SIZE, sizeof(uint64_t)) / sizeof(uint64_t)];
-static nl::Weave::PersistedCounter sCritEventIdCounter;
-
 uint64_t gProdEventBuffer[RoundUp(WEAVE_DEVICE_CONFIG_EVENT_LOGGING_PROD_BUFFER_SIZE, sizeof(uint64_t)) / sizeof(uint64_t)];
-static nl::Weave::PersistedCounter sProdEventIdCounter;
-
 #if WEAVE_DEVICE_CONFIG_EVENT_LOGGING_INFO_BUFFER_SIZE
 uint64_t gInfoEventBuffer[RoundUp(WEAVE_DEVICE_CONFIG_EVENT_LOGGING_INFO_BUFFER_SIZE, sizeof(uint64_t)) / sizeof(uint64_t)];
-static nl::Weave::PersistedCounter sInfoEventIdCounter;
 #endif
-
 #if WEAVE_DEVICE_CONFIG_EVENT_LOGGING_DEBUG_BUFFER_SIZE
 uint64_t gDebugEventBuffer[RoundUp(WEAVE_DEVICE_CONFIG_EVENT_LOGGING_DEBUG_BUFFER_SIZE, sizeof(uint64_t)) / sizeof(uint64_t)];
-static nl::Weave::PersistedCounter sDebugEventIdCounter;
 #endif
+#endif
+
+static nl::Weave::PersistedCounter sCritEventIdCounter;
+static nl::Weave::PersistedCounter sProdEventIdCounter;
+static nl::Weave::PersistedCounter sInfoEventIdCounter;
+static nl::Weave::PersistedCounter sDebugEventIdCounter;
```

Comment:
These need to be inside #if / #endif blocks so that they are only defined if the corresponding importance level is enabled.